### PR TITLE
[Merged by Bors] - chore: move to v4.5.0-rc1, and merge changes from `bump/v4.5.0` branch.

### PR DESCRIPTION
**Pull request by kim-em**
This PR:
* bumps to lean-toolchain to `v4.5.0-rc1`
* bumps the Std and Aesop dependencies to their versions using `v4.5.0-rc1`
* merge the already reviewed changes from the `bump/v4.5.0` branch
  * adaptations for leanprover/lean4#2923 in #9011
  * adaptations for leanprover/lean4#2973 in #9161
  * adaptations for leanprover/lean4#2964 in #9176

### CODE REVIEW

File: test/abel.lean [hunk 1]
```diff
@@ -98,15 +98,8 @@ error: abel_nf made no progress
...
 
-/--
-error: no goals to be solved
--/
--- This error message is confusing: it is saying that it closed the main goal,
--- and so then had nothing to do on the hypotheses.
--- The user has to guess that they should remove the `at *`.
-#guard_msgs in
 example [AddCommGroup α] (x y z : α) (_w : x = y + z) : x - x = 0 := by
-  abel_nf at *
```
Comment (eric-wieser) [inline test/abel.lean:109]: As commented in https://github.com/leanprover-community/mathlib4/pull/9011/files#r1426012984, this looks a bit odd; I think this test was intended to test the error message for `at *`, not to succeed.

Comment (semorrison) [reply test/abel.lean:109]: The context here is https://github.com/leanprover/lean4/pull/2917.

The test was previously checking for an error message which we decided shouldn't be produced at all!

Comment (semorrison) [reply test/abel.lean:109]: I've changed the tactic back to `abel_nf at *`, and added a comment explaining the history.

**Comment by PatrickMassot** [comment]
bors d+

**Comment by semorrison** [comment]
bors p=10

bors merge

**Pull request closed without merging**